### ISSUE TITLE
Not able to set up GCP CloudSQL as persistent data layer with CloudRun

### ISSUE THREAD

**Issue by jzhan6688** [heartexlabs/label-studio#2729]
**Describe the bug**
Hi team - With default GCP CloudRun one-button deployment, the metadata in SQL-Lite Database will be gone from time to time. So I followed LabelStudio [doc](https://labelstud.io/guide/storedata.html#PostgreSQL-database) to create PostgreSQL in GCP for data persistence.  This is how I did it:

(1) Set up PostgreSQL in GCP CloudSQL
(2) Set up env vars and connection in GCP CloudRun and deploy manually

The CloudRun instance could be initialized successfully but the metadata(project, user.., etc) is not persistent. I used BigQuery to connect to CloudSQL and checked the tables there. It seems the metadata is not written in CloudSQL PostgresSQL DB. From the log, i can see both sql-lite and postgres. Did I miss anything in set up or is it a bug?

BTW, I checked the issue history in this repo and found related issues:
https://github.com/heartexlabs/label-studio/issues/903
https://github.com/heartexlabs/label-studio/issues/848#issuecomment-827854768
But none of them provide detailed solution for this. I hope my issue ticket could be the example for many other engineers who have the same issue.

I am pasting all the details as screenshots as below.

**Expected behavior**
I am expecting to see persistent metadata after I set up PostgreSQL in GCP with CloudRun and CloudSQL

**Screenshots**
CloudSQL setup
![image](https://user-images.githubusercontent.com/101445799/180849139-5290e97c-5110-416a-b65c-f1bcad722760.png)
![image](https://user-images.githubusercontent.com/101445799/180849171-1208624c-c9ce-4cbd-a088-fe67f73712e5.png)

CloudRun setup
![Screen Shot 2022-07-25 at 11 35 16 AM](https://user-images.githubusercontent.com/101445799/180849712-c90bfa81-6f74-451f-919a-5902e9612482.png)

CloudRun log
![image](https://user-images.githubusercontent.com/101445799/180931994-aae45ee8-10d5-4ba1-b04a-bf8648e2ef53.png)
![Screen Shot 2022-07-25 at 10 53 32 PM](https://user-images.githubusercontent.com/101445799/180933427-cc512b6a-6ff4-42c3-b62c-4af66e5693e7.png)

BigQuery query:
![Screen Shot 2022-07-25 at 11 38 25 AM](https://user-images.githubusercontent.com/101445799/180850218-61d5d45f-0576-4d92-a315-2ad58e0ea4b4.png)

**Environment (please complete the following information):**
 - OS: [Mac OS with Chrome]
 - Label Studio Version [1.5.0]

**Comment by farioas** [comment]
Cloud-run-button has no support for provisioning of dependent services:
https://github.com/GoogleCloudPlatform/cloud-run-button/issues/74
https://github.com/GoogleCloudPlatform/cloud-run-button/issues/129

According to CloudRun documentation, it's not possible to configure direct communication from CloudRun to CloudSQL. You have to provision Cloud SQL proxy, this can be easily done in the tab "Connections"
![180849712-c90bfa81-6f74-451f-919a-5902e9612482](https://user-images.githubusercontent.com/2744993/180966448-e4e5b34f-c695-4e0c-8b75-28d6eacad5d5.png)

Documentation: https://cloud.google.com/sql/docs/postgres/connect-run#configure

I agree, this should be covered by our documentation, but we're working on something special to cover all these.

**Comment by jzhan6688** [comment]
> Cloud-run-button has no support for provisioning of dependent services: [GoogleCloudPlatform/cloud-run-button#74](https://github.com/GoogleCloudPlatform/cloud-run-button/issues/74) [GoogleCloudPlatform/cloud-run-button#129](https://github.com/GoogleCloudPlatform/cloud-run-button/issues/129)
> 
> According to CloudRun documentation, it's not possible to configure direct communication from CloudRun to CloudSQL. You have to provision Cloud SQL proxy, this can be easily done in the tab "Connections" ![180849712-c90bfa81-6f74-451f-919a-5902e9612482](https://user-images.githubusercontent.com/2744993/180966448-e4e5b34f-c695-4e0c-8b75-28d6eacad5d5.png)
> 
> Documentation: https://cloud.google.com/sql/docs/postgres/connect-run#configure
> 
> I agree, this should be covered by our documentation, but we're working on something special to cover all these.

Thx @farioas - I read through the doc you pasted and also this [example](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/main/cloud-sql/postgres/sqlalchemy). So I added a new env var in `Connections` tab as follows
![Screen Shot 2022-07-26 at 11 25 52 AM](https://user-images.githubusercontent.com/101445799/181084132-57c76177-71e7-4226-b872-b9b2876f9eba.png). But I still couldn't see data persistence. Did I still miss anything? Thx!

**Comment by farioas** [comment]
Looks like you missed step 4:
> Connect using your instance's private IP and port 5432.

Since you provisioned a proxy it's available from within CloudRun container, so you should override POSTGRE_HOST:
```
POSTGRE_HOST=127.0.0.1
```

INSTANCE_UNIX_SOCKET is useless here, you can safely remove it, it's only supported in example that you mentioned.

**Comment by jzhan6688** [comment]
Hi @farioas  - thx for the reply! I am using public IP configuration and that's why I didn't set host to be private IP. Does that mean we have to use private IP for LabelStudio deployment? Thx!

![image](https://user-images.githubusercontent.com/101445799/181121050-63bb47f9-93db-4d1b-80fa-90b8b67d84d2.png)

**Comment by farioas** [comment]
You're right, you don't need to use localhost for Public IP.
Just tested your setup, after adding this variable everything works fine.
```
DJANGO_DB=default
```

Btw, CloudRun was designed for stateless services. All files uploaded by direct Import into LS will be wiped after container restart, that happens frequently (few times a day). That's ok for PoC purposes, but for long term purposes it's better to use external storage: https://labelstud.io/guide/storage.html

**Comment by farioas** [comment]
Also, I just discovered a bug in CloudRun. The fix will be ready in the next 1 weeks.
You may see this window when you'll try to create a new project, just close it and refresh the page.
<img width="724" alt="Screenshot 2022-07-27 at 18 51 10" src="https://user-images.githubusercontent.com/2744993/181281314-7cb48c52-6773-4b8e-a4ea-7f87c1bce8e3.png">

**Comment by jzhan6688** [comment]
Thank you @farioas ! After I added `DJANGO_DB=default `, I can see new project and new user in CloudSQL through BigQuery. I think I already set up the cloud storage sync in projects. 

After container in CloudRun re-starts, the metadata regarding project/users etc should still be there after we set this data persistence layer, right?

Also regarding the error you pasted, I met that before - it seems current version of LS requires us to import an image to create a new project. If we want to create an empty project without task, this error is thrown out. Yeah, I agree, this could be a feature improvement for next version.
